### DISCUSSION TITLE
A Setting to only allow equally named vaults

### DISCUSSION THREAD

**Discussion by SpaceMoehre** [vrtmrz/obsidian-livesync#299] [Ideas]
When working, it needs only one sleepy coworker, that syncs another vault with the main vault. I'd love a setting were you can enable a check for the vaults name to be equivalent to the other vaults and when its different, then the plugin errors, thus preventing someone from accidentally leaking information from other projects, having to clean and filling the couchdb.

**Comment by vrtmrz** [comment]
Thank you for the idea! However, I think it probably could not be resolved by a simple name-checking. 
We can make vaults which are named the same and we can configure the local database name in these cases.
Therefore we may need to configure the expected remote database name.

And, some people use some tricky name databases; I also use one which is suffixed with a generation number for keeping them as backups. (e.g., `main` vault syncs with `main202309` now).

I suppose this idea is intended to warn at the initial replication to the new remote server. It can be detectable, therefore, how about following behaviour?

--- 
**Notice**
We are connecting to the new or rebuilt database.
Are you sure to synchronise the vault "main" to the database "http://example.com/main_2"?

\[ YES, synchronise! \] \[CANCEL\] 

--- 

If this dialogue works also in your case, we do not have to make a new toggle (It is distressing that there are so many toggles that you all have to worry about).
May I have your opinion, please?
